src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitSchedulerClient.kt

```diff
@@ -0,0 +1,130 @@
+package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.client
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+import org.springframework.web.util.UriBuilder
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.ChangeVisitSlotRequestDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.ReserveVisitSlotDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.RestPage
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.SupportTypeDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.VisitCancelDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.VisitDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.VisitSearchDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.VisitSessionDto
+import java.time.Duration
+import java.util.Optional
+
+@Component
+class VisitSchedulerClient(
+
+  @Qualifier("visitSchedulerWebClient") private val webClient: WebClient,
+  @Value("\${visit-scheduler.api.timeout:10s}") val apiTimeout: Duration
+) {
+  private val pagedModel = object : ParameterizedTypeReference<RestPage<VisitDto>>() {}
+
+  fun getVisitByReference(reference: String): VisitDto? {
+    return webClient.get()
+      .uri("/visits/$reference")
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono<VisitDto>().block(apiTimeout)
+  }
+
+  fun getVisits(visitSearchDto: VisitSearchDto): RestPage<VisitDto>? {
+    return webClient.get()
+      .uri("/visits/search") {
+        visitSearchUriBuilder(visitSearchDto, it).build()
```

Comment:
It's not a DTO because it's not being transported... you have 4 choices here!
1. You just pass around the parameters the same as the controller.
2. You change the name to VisitSearchBean
3. use @RequestParam Map<String,String> params
4. http://dolszewski.com/spring/how-to-bind-requestparam-to-object/ maps params to object attribs..

Reply:
Went for option 2 and moved it to a new package (filter) and renamed it to VisitSearchRequestFilter. Also looked at Option and it works well. The only issue is that the Swagger document does not show it as @requestParam. I will raise a tech debt ticket to look at implementing Option 4 with the correct Swagger documentation and will pick it up later when working on tech debt.